src/pages/rt-cor-group/forms/CorrespondentGroupForm.js

```diff
@@ -0,0 +1,113 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { useForm } from 'src/hooks/form'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { RemittanceSettingsRepository } from 'src/repositories/RemittanceRepository'
+
+export default function CorrespondentGroupForm({ labels, maxAccess, recordId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: RemittanceSettingsRepository.CorrespondentGroup.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      reference: '',
+      name: ''
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      reference: yup.string().required(),
+      name: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: RemittanceSettingsRepository.CorrespondentGroup.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!obj.recordId) {
+          toast.success(platformLabels.Added)
+          formik.setValues({
```

Comment:
no need for destructuring , replace by `fomrik.setFieldValue('recordId',response.recordId)`